skipruntime-ts/adapters/postgres/src/index.ts

```diff
@@ -108,15 +112,21 @@ export class PostgresExternalService implements ExternalService {
     };
 
     const initData = async () => {
-      callbacks.loading();
-      const init = await this.client.query(format("SELECT * FROM %I;", table));
-      const entries: Map<Json, Json[]> = new Map<Json, Json[]>();
-      for (const row of init.rows as { [col: string]: Json }[]) {
-        const k = row[key.col]!;
-        if (entries.has(k)) entries.get(k)!.push(row);
-        else entries.set(k, [row]);
+      if (params.ignoreOldData) {
+        callbacks.update([], true);
+      } else {
+        callbacks.loading();
+        const init = await this.client.query(
+          format("SELECT * FROM %I;", table),
```

Comment:
I wonder if there is a small bool->int generalization possible here, where instead of syncing all or none of the historic data, there is a number that could be put as a limit on this query. Not sure it is better, but maybe.